Tests/DatadogTests/Datadog/LoggerTests.swift

```diff
@@ -186,16 +186,16 @@ class LoggerTests: XCTestCase {
         Datadog.setUserInfo(id: "abc-123", name: "Foo")
         logger.debug("message with user `id` and `name`")
 
-        Datadog.setUserInfo(id: "abc-123", name: "Foo", email: "foo@example.com")
-        logger.debug("message with user `id`, `name` and `email`")
+        Datadog.setUserInfo(id: "abc-123", name: "Foo", email: "foo@example.com", extraInfo: ["extra_key": "extra_value"])
```

Comment:
Because `extraInfo` encoding depends on the feature (Logging and RUM supports multiple `Encodable` types, but Tracing encodes all values as `String`), it seems better to mix few types in this test:
```swift
extraInfo: ["extra_string": "string_value", "extra_int": 42, "extra_bool": true]
```
and document the expected encoding result in test assertions.

Reply:
i can't pass different types in a dict to `assertUserInfo` method below
i will need to use `matcher.assertValue(forKey:value:)`